doc/rst/language/spec/procedures.rst

```diff
@@ -586,95 +586,51 @@ modify the formal argument within its dynamic scope. Whether ``const``
 is interpreted as ``const in`` or ``const ref`` intent depends on the
 argument type.  Generally, small values, such as scalar types, will be
 passed by ``const in``; while larger values, such as domains and
-arrays, will be passed by ``const ref`` intent. The
-:ref:`Abstract_Intents_Table` below lists the meaning of the ``const``
-intent for each type.
+arrays, will be passed by ``const ref`` intent.
+
+For the following types, the ``const`` intent means ``const in``:
+
+  * scalar types (``bool``, ``int``, ``uint``, ``real``, ``imag``, ``complex``)
+  * ranges
+  * non-managed classes (``borrowed``, ``unmanaged``)
+
+For the following types, the ``const`` intent means ``const ref``:
+
+  * string-like types (``string``, ``bytes``)
+  * domains / domain maps
+  * arrays
+  * records
+  * auto-managed classes (``owned``, ``shared``) - see :ref:`Default_Intent_for_owned_and_shared`
+  * unions ``const ref``
+  * synchronization types (``atomic``, ``sync``)
+
+The ``const`` intent for tuples applies the ``const`` intent to each tuple
+component as if it was passed as a separate argument.
+See :ref:`Tuple_Argument_Intents`.
 
 .. _The_Default_Intent:
 
 The Default Intent
 ^^^^^^^^^^^^^^^^^^
 
 When no intent is specified for a formal argument, the *default
-intent* is applied.  It is designed to take the most natural/least
-surprising action for the argument, based on its type.  In practice,
+intent* is applied. It is designed to take the most natural/least
+surprising action for the argument, based on its type. In practice,
 this is ``const`` for most types (as defined by
 :ref:`The_Const_Intent`) to avoid surprises for programmers coming
 from languages where everything is passed by ``in`` or ``ref`` intent
-by default.  Exceptions are made for types where modification is
-considered part of their nature, such as types used for synchronization
-(like ``atomic``) and arrays.
+by default. Exceptions are made for types where modification is considered part
+of their nature, such as types used for synchronization
+(like ``atomic`` or ``sync``).
+
+For the following types, the default intent is not ``const``:
+
+  * synchronization types (``atomic``, ``sync``) - the default intent is ``ref``
```

Comment:
This feels a bit redundant with the previous sentence. My recommendation is to keep this (because I like the way the space here calls more attention to the detail) and simplify the previous sentence.

Suggestion:

-new paragraph-
Exceptions are made for types where modification is considered part
of their nature. In particular, the the default intent for the following types is ``ref``:
  * ``atomic``
  * ``sync``